app/src/main/java/com/asfoundation/wallet/change_currency/bottom_sheet/ChooseCurrencyBottomSheetViewModel.kt

```diff
@@ -0,0 +1,45 @@
+package com.asfoundation.wallet.change_currency.bottom_sheet
+
+import android.util.Log
+import com.asfoundation.wallet.base.Async
+import com.asfoundation.wallet.base.BaseViewModel
+import com.asfoundation.wallet.base.SideEffect
+import com.asfoundation.wallet.base.ViewState
+import com.asfoundation.wallet.change_currency.use_cases.SetSelectedCurrencyUseCase
+import io.reactivex.Completable
+import io.reactivex.Scheduler
+
+
+sealed class ChooseCurrencyBottomSideEffect : SideEffect {
+  object NavigateBack : ChooseCurrencyBottomSideEffect()
+}
+
+data class ChooseCurrencyBottomSheetState(val selectedCurrency: String,
+                                          val selectedFlag: String,
+                                          val selectedLabel: String,
+                                          val selectedConfirmationAsync: Async<Unit> = Async.Uninitialized) :
+    ViewState
+
+class ChooseCurrencyBottomSheetViewModel(data: ChooseCurrencyBottomSheetData,
+                                         private val networkScheduler: Scheduler,
+                                         private val setSelectedCurrencyUseCase: SetSelectedCurrencyUseCase) :
+    BaseViewModel<ChooseCurrencyBottomSheetState, ChooseCurrencyBottomSideEffect>(
+        initialState(data)) {
+
+  companion object {
+    fun initialState(data: ChooseCurrencyBottomSheetData): ChooseCurrencyBottomSheetState {
+      return ChooseCurrencyBottomSheetState(data.currency!!, data.flag!!, data.label!!)
+    }
+  }
+
+  fun currencyConfirmationClick() {
+    Completable.fromAction { setSelectedCurrencyUseCase(state.selectedCurrency) }
```

Comment:
We talked about this at the time, but I think it is cleaner if the use case is a Completable instead. I guess every UseCase generally should be contained within an Rx chain, since it is the framework we use throughout the architecture stack.